### ISSUE TITLE
[Bug]: Menu - secondary content is white instead of black in HC theme on hover

### ISSUE THREAD

**Issue by jurokapsiar** [microsoft/fluentui#31912]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
any system in the universe
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://react.fluentui.dev/?path=/docs/components-menu-menu--default&globals=storybook_fluentui-react-addon_theme:teams-high-contrast#secondary-content-for-menu-items

## Actual Behavior
<img width="229" alt="image" src="https://github.com/microsoft/fluentui/assets/13742664/9f739758-0601-4c6c-82d8-015714c0c53e">

## Expected Behavior
<img width="250" alt="image" src="https://github.com/microsoft/fluentui/assets/13742664/4e7f7ae6-b5d8-410c-8da8-74d9d8086813">

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
